pkg/controller/migdirect/rsync.go

```diff
@@ -0,0 +1,827 @@
+func (t *Task) createRsyncClientPods() error {
+	// Get client for destination
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	pvcMap := t.getPVCNamespaceMap()
+	for ns, vols := range pvcMap {
+		// Get stunnel svc IP
+		svc := corev1.Service{}
+		key := types.NamespacedName{Name: "migdirect-rsync-transfer-svc", Namespace: ns}
+		srcClient.Get(context.TODO(), key, &svc)
+		ip := svc.Spec.ClusterIP
+
+		trueBool := true
+		runAsUser := int64(0)
+
+		// Add PVC volume mounts
+		for _, vol := range vols {
+			volumes := []corev1.Volume{}
+			volumeMounts := []corev1.VolumeMount{}
+			containers := []corev1.Container{}
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      vol,
+				MountPath: fmt.Sprintf("/mnt/%s/%s", ns, vol),
+			})
+			volumes = append(volumes, corev1.Volume{
+				Name: vol,
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: vol,
+					},
+				},
+			})
+			containers = append(containers, corev1.Container{
+				Name:  fmt.Sprintf("rsync-client-%s", vol),
+				Image: "quay.io/konveyor/rsync-transfer:latest",
+				Env: []corev1.EnvVar{
+					{
+						Name:  "RSYNC_PASSWORD",
+						Value: "changeme",
+					},
+				},
+				Command: []string{"rsync", "-aPvvHh", "--delete", "--port", "2222", "--log-file", "/dev/stdout", fmt.Sprintf("/mnt/%s/%s/", ns, vol), fmt.Sprintf("rsync://root@%s/%s", ip, vol)},
+				Ports: []corev1.ContainerPort{
+					{
+						Name:          "rsync-client",
+						Protocol:      corev1.ProtocolTCP,
+						ContainerPort: int32(22),
+					},
+				},
+				VolumeMounts: volumeMounts,
+				SecurityContext: &corev1.SecurityContext{
+					Privileged:             &trueBool,
+					RunAsUser:              &runAsUser,
+					ReadOnlyRootFilesystem: &trueBool,
+				},
+			})
+			clientPod := corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("migdirect-rsync-transfer-%s", vol),
+					Namespace: ns,
+					Labels: map[string]string{
+						"app":       "migdirect-rsync-transfer",
+						"migdirect": "rsync-client",
+					},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes:       volumes,
+					Containers:    containers,
+				},
+			}
+			err = srcClient.Create(context.TODO(), &clientPod)
+			if k8serror.IsAlreadyExists(err) {
+				t.Log.Info("Rsync client pod already exists on source", "namespace", clientPod.Namespace)
+			} else if err != nil {
+				return err
+			}
+			t.Log.Info("Rsync client pod created", "name", clientPod.Name, "namespace", clientPod.Namespace)
+		}
+
+	}
+	return nil
+}
+
+func (t *Task) haveRsyncClientPodsCompleted() (bool, error) {
+	// Get client for destination
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return false, err
+	}
+	selector := labels.SelectorFromSet(map[string]string{
+		"app":       "migdirect-rsync-transfer",
+		"migdirect": "rsync-client",
```

Comment:
Same comment here about using correlation labels instead of generically tying back to all MigDirect CRs